tests/client_api.cpp

```diff
@@ -82,6 +82,24 @@ TEST(ClientAPI, LoginWrongUsername)
         mtx_client->close();
 }
 
+TEST(ClientAPI, ChangeDisplayName)
+{
+        std::shared_ptr<Client> mtx_client = std::make_shared<Client>("localhost");
+
+        mtx_client->login(
+          "alice", "secret", [mtx_client](const mtx::responses::Login &res, ErrType err) {
+                  ASSERT_FALSE(err);
+                  validate_login("@alice:localhost", res);
+
+                  // Change the display name to Arthur Dent and verify its success through the lack
+                  // of an error
+                  mtx_client->set_displayname("Arthur Dent",
```

Comment:
Could you also add a call where the displayname is an empty string?

Reply:
Added and it didn't work at first, but now it does. I needed to change the DisplayName to_json function in matrix-structs to serialize whether the string is empty or not in order for this empty string test to work.